USF-2163 Storefront Builder documentation (#338)
commerce-docs/microsite-commerce-storefront: src/content/docs/merchants/storefront-builder/index.mdx

---
title: About Storefront Builder
description: Learn about the Storefront Builder and how to use it to build your storefront.
---

import { Steps } from '@astrojs/starlight/components';

**Storefront Builder** is included with Adobe Commerce as a Cloud Service, and is designed to streamline both the technical and editorial workflows for building and maintaining commerce storefronts. 

## What is the Storefront Builder?

The **Storefront Builder** is a collection of tools and technologies from Adobe Commerce designed to make it easy to create, manage, and customize modern, high-performance storefronts. It brings together the following elements of a commerce storefront:

- **DA.live (Document Authoring Live)**: The content management system where your storefront's pages, blocks, and assets are authored and managed.

> - **Content and Commerce Blocks**: Pre-built, reusable UI components (like carts, product lists, etc.) that can be easily added to your storefront.

Reviewer: ```suggestion
- **Content and Commerce Blocks**: Pre-built, reusable UI components (carts, product lists, etc.) that can be seamlessly integrated into your storefront.
```

### Edge Delivery Services (EDS)

Edge Delivery Services (EDS) is a fast, scalable platform for delivering your storefront's content and assets to customers. EDS leverages a global content delivery network (CDN) to ensure that pages, images, and other resources load quickly, no matter where your customers are located. By serving content from the edge, EDS reduces latency and improves site performance, which is critical for both user experience and SEO. EDS is tightly integrated with the Storefront Builder ecosystem, ensuring seamless deployment and updates of your storefront content.

### Content and Commerce Blocks

> Drop-ins and Blocks are pre-built, reusable UI components that can be easily added to your storefront. Examples include shopping carts, product lists, navigation menus, and more. These components are designed to accelerate storefront development by providing ready-made, best-practice implementations for common commerce features. Drop-ins are typically more dynamic and commerce-aware, while blocks are often used for layout and content structuring. Both can be configured and extended to match your brand and business needs, and are fully compatible with localization and multistore setups.

Reviewer: ```suggestion
Blocks are a foundational concept for adding form and function to page sections. They are defined using tables in the content and decorated by your custom code during page rendering. Drop-ins are dynamic, commerce-specific front-end components that can be integrated within a Block.
```

### Product Visuals (AEM Assets)

Product Visuals is the digital asset management (DAM) solution used for storing and managing images, videos, and other media assets for your storefront. With Product Visuals, you can organize your media library, apply metadata and tags, and ensure that the right assets are used in the right places across your storefront. Product Visuals integrates with DA.live and Visual Editor, making it easy to insert and update media content as part of your authoring workflow. It also works seamlessly with EDS to deliver optimized assets to customers around the world, ensuring fast load times and high-quality visuals.

## How to set up a new storefront

There are several ways to set up a new storefront:

- **CLI**: The Storefront Builder provides a CLI (Command Line Interface) tool that automates the setup and configuration of a new Edge Delivery Commerce storefront.
- **Site Creator UI**: A web-based tool that guides you through the setup process with a user interface, for those who prefer not to use the command line.

> - **Manual Setup**: You can also follow step-by-step documentation to set up your storefront manually, which is useful for learning or for custom setups.

Reviewer: (had a comment I couldn't delete)

## What is the Storefront Builder?

The **Storefront Builder** is a collection of tools and technologies from Adobe Commerce designed to make it easy to create, manage, and customize modern, high-performance storefronts. It brings together the following elements of a commerce storefront:

- **DA.live (Document Authoring Live)**: The content management system where your storefront's pages, blocks, and assets are authored and managed.
- **Content and Commerce Blocks**: Pre-built, reusable UI components (carts, product lists, etc.) that can be seamlessly integrated into your storefront.
- **Edge Delivery Services (EDS)**: A fast, scalable delivery platform for your storefront content and assets.
- **Visual Editor**: A visual, in-context editing tool that lets business users and content authors update storefront content directly, with real-time preview.
- **Digital Asset Management (DAM)**: For storing and managing editorial assets like images and other media used on content pages and product enrichments in your storefront.

### DA.live (Document Authoring Live)

> DA.live (Document Authoring) is the content management system (CMS) at the heart of Storefront Builder. It is where your storefront's pages, blocks, and assets are authored, organized, and managed. DA.live provides a collaborative environment for content teams to create and structure site content, manage translations, and control publishing workflows. It integrates with Visual Editor for in-context editing and with EDS for fast, reliable content delivery. DA.live supports versioning, localization, and multistore management, making it a powerful tool for complex commerce operations.

Reviewer 2: ```suggestion
DA.live (Document Authoring) is the content management system (CMS) at the heart of Storefront Builder. It is where your storefront's pages, blocks, and assets are authored, organized, and managed. DA.live provides a collaborative environment for content teams to create and structure site content, manage translations, and control publishing workflows. It integrates with Visual Editor for in-context editing and with Edge Delivery Services for fast, reliable content delivery. DA.live supports versioning, localization, and multistore management, making it a powerful tool for complex commerce operations.
```

### Edge Delivery Services (EDS)

> Edge Delivery Services (EDS) is a fast, scalable platform for delivering your storefront's content and assets to customers. EDS leverages a global content delivery network (CDN) to ensure that pages, images, and other resources load quickly, no matter where your customers are located. By serving content from the edge, EDS reduces latency and improves site performance, which is critical for both user experience and SEO. EDS is tightly integrated with the Storefront Builder ecosystem, ensuring seamless deployment and updates of your storefront content.

Reviewer 2: I don't think we're supposed to abbreviate Edge Delivery Services.

> Edge Delivery Services (EDS) is a fast, scalable platform for delivering your storefront's content and assets to customers. EDS leverages a global content delivery network (CDN) to ensure that pages, images, and other resources load quickly, no matter where your customers are located. By serving content from the edge, EDS reduces latency and improves site performance, which is critical for both user experience and SEO. EDS is tightly integrated with the Storefront Builder ecosystem, ensuring seamless deployment and updates of your storefront content.

Reviewer 2: ```suggestion
Edge Delivery Services is a fast, scalable platform for delivering your storefront's content and assets to customers. EDS leverages a global content delivery network (CDN) to ensure that pages, images, and other resources load quickly, no matter where your customers are located. By serving content from the edge, Edge Delivery Services reduces latency and improves site performance, which is critical for both user experience and SEO. Edge Delivery Services is tightly integrated with the Storefront Builder ecosystem, ensuring seamless deployment and updates of your storefront content.
```

### Content and Commerce Blocks

Blocks are a foundational concept for adding form and function to page sections. They are defined using tables in the content and decorated by your custom code during page rendering. Drop-ins are dynamic, commerce-specific front-end components that can be integrated within a Block.

### Visual Editor (Universal Editor features)

Visual Editor is a visual, in-context editing tool that empowers business users and content authors to update storefront content directly on the page, with real-time preview. With Visual Editor, you can make changes to text, images, blocks, and drop-ins without needing to write code or wait for developer assistance. This tool streamlines content workflows, reduces time-to-market for updates, and ensures that your storefront always reflects the latest campaigns, promotions, and brand messaging. Visual Editor is fully integrated with DA.live and supports localization and multistore editing.

### Product Visuals

> Product Visuals is an integration of product assets into the ACCS backend. For Storefront Builder, the digital asset management (DAM) solution accesseseditorial assets like images, videos, and other media used on content pages and product enrichments. The DAM integrates with DA.live and Visual Editor for seamless media content management and works with EDS to deliver optimized assets globally.

Reviewer 2: ```suggestion
[Product Visuals](https://experienceleague.adobe.com/en/docs/commerce/product-visuals/overview) is an integration of Adobe Experience Manager Assets with your Adobe Commerce backend. For Storefront Builder, the digital asset management (DAM) solution accesses editorial assets like images, videos, and other media used on content pages and product enrichments. The DAM integrates with DA.live and Visual Editor for seamless media content management and works with Edge Delivery Services to deliver optimized assets globally.
```
